apps/beeswax/src/beeswax/server/hive_server2_lib.py

```diff
@@ -989,7 +989,10 @@ def execute_async_query(self, query, statement=0, session=None):
 
     # The query can override the default configuration
     configuration.update(self._get_query_configuration(query))
-    query_statement = query.get_query_statement(statement)
+    if HPLSQL.get() and self.query_server['server_name'] == 'beeswax':
+      query_statement = query.hql_query
+    else:
+      query_statement = query.get_query_statement(statement)
```

Comment:
Could the logic be contained inside `get_query_statement()`?

Reply:
same as above.